app/services/post_alerter.rb

```diff
@@ -37,6 +37,10 @@ def self.create_notification_alert(user:, post:, notification_type:, excerpt: ni
   def self.push_notification(user, payload)
     return if user.do_not_disturb?
 
+    return unless DiscoursePluginRegistry.push_notification_filters.all? do |filter|
+      filter.call(user, payload)
+    end
```

Comment:
Just a thought but I was wondering if we can just return immediately within the block once a filter returns `false`.